reconcile/utils/terrascript_aws_client.py

```diff
@@ -5321,16 +5321,27 @@ def populate_tf_resource_rosa_authenticator(self, spec):
         )
         tf_resources.append(cognito_resource_server_gateway_resource)
 
-        # POOL CLIENT
-        cognito_user_pool_client = aws_cognito_user_pool_client(
+        # OCM POOL CLIENT
+        ocm_cognito_user_pool_client = aws_cognito_user_pool_client(
             "userpool_client",
             name=f"ocm-{identifier}-pool-client",
             user_pool_id=f"${{{cognito_user_pool_resource.id}}}",
             callback_urls=[f"{bucket_url}/token.html"],
             depends_on=["aws_cognito_resource_server.userpool_gateway_resource_server"],
             **pool_client_args,
         )
-        tf_resources.append(cognito_user_pool_client)
+        tf_resources.append(ocm_cognito_user_pool_client)
+
+        # INSIGHTS POOL CLIENT
+        insights_cognito_user_pool_client = aws_cognito_user_pool_client(
+            "ins_userpool_client",
+            name=f"insights-{identifier}-pool-client",
+            user_pool_id=f"${{{cognito_user_pool_resource.id}}}",
+            callback_urls=[f"{bucket_url}/token.html"],
+            depends_on=["aws_cognito_resource_server.userpool_gateway_resource_server"],
```

Comment:
@lphiri do we need this dependency here?

Reply:
We do not.  Our API will not have any customer resources or custom roles.